### PR TITLE
Remove --allow-unauthenticated from apt-get call.

### PR DESCRIPTION
**Pull request by obi1kenobi**
Adds the public key matching the docker-engine package, and authenticates the package during installation.

Addresses #173

### CODE REVIEW

File: phase1/aws-ec2-terraform/secure-v1.2-user-data.yaml [hunk 1]
```diff
@@ -197,7 +197,8 @@ write_files:
 
       systemctl daemon-reload
 
-      docker -v || env DEBIAN_FRONTEND=noninteractive apt-get install -y --allow-unauthenticated --force-yes  docker-engine
+      apt-key adv --recv-keys --keyserver keys.ubuntu.com F76221572C52609D
```
Comment (errordeveloper) [inline phase1/aws-ec2-terraform/secure-v1.2-user-data.yaml]: Is this not the same key that we already have above?

https://github.com/obi1kenobi/kubernetes-anywhere/blob/7191292c35bba7c11685dda90b97ff38b564ac96/phase1/aws-ec2-terraform/secure-v1.2-user-data.yaml#L20-L50

**Comment by errordeveloper** [comment]
I don't think this fixes the problem with installing Docker package. Have you tested this?

**Comment by obi1kenobi** [comment]
The following worked on a fresh Ubuntu 16.04 VM:
```
$ sudo add-apt-repository "deb https://apt.dockerproject.org/repo ubuntu-xenial main"
$ sudo apt-key adv --recv-keys --keyserver keyserver.ubuntu.com F76221572C52609D
$ sudo apt-get update
$ sudo apt-get install docker-engine
< docker-engine installs with no errors >
```

In the issue you mentioned in #173, someone pointed out that they also got this sequence of commands to work, although other sequences did not. Also, this repository's tests passed.

**Comment by obi1kenobi** [comment]
Also, while we're at it, thoughts on installing `apt-transport-https` and `ca-certificates` as well?

**Comment by obi1kenobi** [comment]
The entire `aws-ec2-terraform` folder appears to have been removed from the repo, so I'm closing this PR and its underlying issue.

**Pull request closed without merging**